source/ed/connectors/fusio_parser.cpp

```diff
@@ -1276,6 +1277,77 @@ void CommentLinksFusioHandler::handle_line(Data&, const csv_row& row, bool) {
     }
 }
 
+void ObjectCodesFusioHandler::init(Data&){
+    object_uri_c = csv.get_pos_col("object_id");
+    object_type_c = csv.get_pos_col("object_type");
+    code_c = csv.get_pos_col("object_code");
+    object_system_c = csv.get_pos_col("object_system");
+}
+
+
+void ObjectCodesFusioHandler::handle_line(Data& data, const csv_row& row, bool) {
+    if(! has_col(object_uri_c, row) || ! has_col(object_type_c, row) || ! has_col(code_c, row) || ! has_col(object_system_c, row)) {
+        LOG4CPLUS_FATAL(logger, "Error while reading " + csv.filename +
+                        "  impossible to find all needed fields");
+        throw InvalidHeaders(csv.filename);
+    }
+
+    std::string key = boost::algorithm::to_lower_copy(row[object_system_c]);
+    std::string object_type = boost::algorithm::to_lower_copy(row[object_type_c]);
+
+    if (key == "navitia1") {
+        key = "external_code";
+    }
+
+    if (object_type == "line"){
+        const auto& it_object = gtfs_data.line_map.find(row[object_uri_c]);
+        if(it_object != gtfs_data.line_map.end()){
+            data.add_object_code(it_object->second, nt::Type_e::Line, row[code_c], key);
+            if (key == "external_code"){
+                gtfs_data.line_map_by_external_code[row[code_c]] = it_object->second;
+            }
+        }
+        return;
+    }
+    if (object_type == "route"){
+        const auto& it_object = gtfs_data.route_map.find(row[object_uri_c]);
+        if(it_object != gtfs_data.route_map.end()){
+            data.add_object_code(it_object->second, nt::Type_e::Route, row[code_c], key);
+        }
+        return;
+    }
+    if (object_type == "network"){
+        const auto& it_object = gtfs_data.network_map.find(row[object_uri_c]);
+        if(it_object != gtfs_data.network_map.end()){
+            data.add_object_code(it_object->second, nt::Type_e::Network, row[code_c], key);
+        }
+        return;
+    }
+    if (object_type == "trip"){
+        const auto& it_object = data.meta_vj_map.find(row[object_uri_c]);
+        if(it_object != data.meta_vj_map.end()){
+            for(const auto& vj : it_object->second.theoric_vj){
+                data.add_object_code(vj, nt::Type_e::VehicleJourney, row[code_c], key);
+            }
+        }
+        return;
+    }
+    if (object_type == "stop_area"){
+        const auto& it_object = gtfs_data.stop_area_map.find(row[object_uri_c]);
+        if(it_object != gtfs_data.stop_area_map.end()){
+            data.add_object_code(it_object->second, nt::Type_e::StopArea, row[code_c], key);
+        }
+    }
```

Comment:
and I prefere the `} else if (object_type == "bob") {` pattern, but that's personal